utils/utils.js

```diff
@@ -171,7 +167,56 @@ const volunteerForNewGiftees = async (user, count) => {
   await batch.commit()
 
   // return the result after the database stuff is complete
-  return { success: result }
+  return { success: "Please refresh the giftee list" }
+}
+
+/**
+ * @param {object} giftee - details about the giftee
+ * @param {string} [giftee.uuid] - UUID of the giftee, if you do not know it
+ * @param {string} [giftee.user] - If the UUID is unknown this is the giftee's uid or user object
+ * @param {object} update - details about the giftee
+ * @param {string} update.field - What part to mark: sent, received, reported
+ * @param {string} [update.message] - If reporting allow a message
+ * @param {boolean} [update.value] - If reporting allow a message
+ * @returns
+ */
+async function markGifteeAccount({uuid, user}, {field, message, value}){
+  // if someone is marking the gift sent they know the uuid of the giftee
+  if(!uuid){
+    if(!user){return {error: "no uuid or user requested"}}
+
+    // if the uuid is undefined then take the user, search for teh account related and return that uuid
```

Comment:
Nice fallback!

Reply:
Ths is less of a fallback and more of being able to deal with this function from both gifter and giftee. (Unless toy are talking about teh double if statement)